### PR TITLE
Fixing APOGEE dataset from duplicates of [O/Fe], [C/Fe] and [N/Fe]

### PR DESCRIPTION
**Pull request by correac**
There is a bug in the APOGEE dataset that the pipeline reads to make the stellar abundance plots. The arrays of [O/Fe], [N/Fe] and [C/Fe] are duplicated. So when the pipeline reads [N/Fe] or [C/Fe], it is actually reading [O/Fe] values. The problem is not in the pipeline script, but in the hdf5 file of the APOGEE data. This was my mistake from a while ago it seems. Big apologies! The fixed APOGEE dataset is included in this pull request.

### CODE REVIEW

**Comment by EvgeniiChaikin** [comment]
@correac thank you for the fix! Can you please also upload the plots showing any simulation vs. APOGEE data before and after the fix? Otherwise, it is not clear what changes this PR brings. (i.e. I do see that the data has been updated, but I can't tell how big the differences are)

**Comment by correac** [comment]
Updated pull request. Main changes:

(1) Modification of the APOGEE data file so that it includes the galactocentric distances from AstroNN added-value catalog.
(2) The galactocentric distances are required in the pipeline-config/stellar_abundances.py script. For that reason the ObservationalData object from velociraptor.observations is not longer used, since it does not support additional data besides the stellar abundances.

To create the APOGEE_data.hdf5 file included in this repository. The following steps were followed:

======

(1) Reading AstroNN data:

input_filename = "apogee_astroNN-DR17.fits"
astroNN = fits.open(input_filename)
astroNN_dataset = astroNN[1].data
astroNN.close()

GalR = astroNN_dataset['galr']
nonan = np.where(np.isnan(GalR) == False)[0]
GalR = GalR[nonan]
Galz = astroNN_dataset['galz'][nonan]
Galphi = astroNN_dataset['galphi'][nonan]
log_g = astroNN_dataset['LOGG'][nonan]
APOGEE_ID_astroNN = astroNN_dataset['APOGEE_ID'][nonan]

=========

(2) Reading APOGEE data:

input_filename = "allStar-l31c.2.fits"
output_directory = "../"

apogee = fits.open(input_filename)
apogee_dataset = apogee[1].data
apogee.close()

sample = (np.bitwise_and(apogee_dataset['aspcapflag'], 2**23) == 0) & (apogee_dataset['extratarg']==0)
selection = np.where(sample)[0]
APOGEE_ID = apogee_dataset['APOGEE_ID'][selection]
FE_H = apogee_dataset['fe_h'][selection]
MG_FE = apogee_dataset['mg_fe'][selection]
N_FE = apogee_dataset['n_fe'][selection]
O_FE = apogee_dataset['o_fe'][selection]
C_FE = apogee_dataset['c_fe'][selection]

sample = (FE_H > -5) & (MG_FE > -5) & (O_FE > -5) & (C_FE > -5) & (N_FE > -5)
selection = np.where(sample)[0]
FE_H = FE_H[selection]
MG_FE = MG_FE[selection]
O_FE = O_FE[selection]
N_FE = N_FE[selection]
C_FE = C_FE[selection]
APOGEE_ID = APOGEE_ID[selection]

======

(3) Cross-match of catalogs using the stars IDs!

_, indx_APOGEE_in_astroNN, indx_astroNN_in_APOGEE = np.intersect1d(APOGEE_ID, APOGEE_ID_astroNN, assume_unique=False, return_indices=True,)

GalR = GalR[indx_astroNN_in_APOGEE]
Galz = Galz[indx_astroNN_in_APOGEE]
Galphi = Galphi[indx_astroNN_in_APOGEE]

FE_H = FE_H[indx_APOGEE_in_astroNN]
MG_FE = MG_FE[indx_APOGEE_in_astroNN]
O_FE = O_FE[indx_APOGEE_in_astroNN]
N_FE = N_FE[indx_APOGEE_in_astroNN]
C_FE = C_FE[indx_APOGEE_in_astroNN]

**Comment by correac** [comment]
APOGEE stellar abundance data in radial and azimuthal cuts.
image

image

image

image

**Review by EvgeniiChaikin — COMMENTED** [review]
Thank you @correac! 

I left one comment about the code.

Also, as discussed earlier today, the current script is a temporary solution and later we will need to update the velociraptor-python library so that it can handle observational datasets as complex as this one, which is a 3D dataset.

**Comment by EvgeniiChaikin** [comment]
@correac After spending some time with the new APOGEE conversion script, I think that the current version is too long for people to be able to easily understand it. For this reason, I wrote a shorter and cleaner version of the script and now would like to push my commit with the changes to your repository. Could you give me access to do it, please?

File: data/StellarAbundances/conversion/convertAPOGEE.py [hunk 1]
```diff
@@ -11,121 +9,126 @@
...
-        y = el_FE
+        x = O_H + O_over_H_GA07 - O_over_H_AS09
+        xlabel = "[O/H]"
+
+        if element == element == "OHMGFE":
+            MG_FE = apogee_dataset["MG_FE"][:]
+            y = MG_FE + Mg_over_Fe_GA07 - Mg_over_Fe_AS09
+            name = "[Mg/Fe] as a function of [O/H]".format(element)
+            ylabel = "[Mg/Fe]"
+        else:  # OH
```
Comment (MatthieuSchaller) [inline data/StellarAbundances/conversion/convertAPOGEE.py:64]: OFe, no?

**Comment by MatthieuSchaller** [comment]
@EvgeniiChaikin two comments from me and then I'll merge.

File: data/StellarAbundances/conversion/convertAPOGEE.py [hunk 2]
```diff
@@ -11,121 +9,126 @@
...
-    elif element == "OHMGFE":
-        x = O_H
-        y = MG_FE
-    else:
-        x = FE_H
-        y = el_FE
+        x = O_H + O_over_H_GA07 - O_over_H_AS09
+        xlabel = "[O/H]"
+
+        if element == element == "OHMGFE":
```
[the remark this answers is not in the document]
Comment (EvgeniiChaikin) [reply data/StellarAbundances/conversion/convertAPOGEE.py]: Agreed! Fixed

File: data/StellarAbundances/conversion/convertAPOGEE.py (hunk 1 above)
Comment (EvgeniiChaikin) [reply data/StellarAbundances/conversion/convertAPOGEE.py]: Done.

**Comment by correac** [comment]
Great changes Evgenii! Anything else I can do for this pull request?

**Review by MatthieuSchaller — APPROVED** [review]

**Pull request merged**